polyply/src/nonbond_engine.py

```diff
@@ -154,12 +165,24 @@ def remove_positions(self, mol_idx, node_key):
         """
         gndx = self.nodes_to_gndx[(mol_idx, node_key)]
         self.positions[gndx] = np.array([np.inf, np.inf, np.inf])
-        self.defined_idxs[-1].remove(gndx)
-        new_tree = scipy.spatial.ckdtree.cKDTree(self.positions[self.defined_idxs[-1]],
-                                                 boxsize=self.boxsize,
-                                                 balanced_tree=False,
-                                                 compact_nodes=False)
-        self.position_trees[-1] = new_tree
+
+        if gndx in self.gndx_to_tree:
+            tree_idx = self.gndx_to_tree[gndx]
+            self.defined_idxs[tree_idx].remove(gndx)
+            new_tree = scipy.spatial.ckdtree.cKDTree(self.positions[self.defined_idxs[tree_idx]],
+                                                     boxsize=self.boxsize,
+                                                     balanced_tree=False,
+                                                     compact_nodes=False)
+            self.position_trees[tree_idx] = new_tree
+            del self.gndx_to_tree[gndx]
+
+    def remove_molecule_positions(self, mol_idx, molecule):
+        """
+        Set all positions of the nodes in molecule
+        to infinity.
```

Comment:
This isn't really correct is it? It just wipes the indices to the trees.
Also, docstring incomplete.

Reply:
I put a more complete docstring and refined this a little. But yeah that is essentially what it does.